client/resources/md5mesh.cpp

```diff
@@ -11,366 +11,376 @@
-					if (line.find("}") != std::string::npos) {
-						this->meshes_internal.push_back(std::move(mesh));
-						break;
+		}
+		else if (identifier == "mesh") {
+			InternalMesh mesh;
+			while (std::getline(f, line)) {
+				std::string line2 = CleanString(line);
+				ss.str(line2);
+				ss.clear();
+				identifier.clear();
+
+				ss >> identifier;
+
+				if (identifier == "shader") {
+					ss >> mesh.shader;
+					auto filename = base_path / mesh.shader;
+					if (!TextureMap::Has(mesh.shader)) {
+						auto pixbuf = PixelBuffer::Create(mesh.shader, filename);
+						auto tex = std::make_shared<TextureObject>(pixbuf);
+						TextureMap::Set(mesh.shader, tex);
 					}
 				}
+				else if (identifier == "numverts") {
+					unsigned int nverts;
+					ss >> nverts;
+					mesh.verts.reserve(nverts);
+				}
+				else if (identifier == "vert") {
+					mesh.verts.push_back(ParseVertex(ss));
+				}
+				else if (identifier == "numtris") {
+					unsigned int ntris;
+					ss >> ntris;
+					mesh.tris.reserve(ntris);
+				}
+				else if (identifier == "tri") {
+					mesh.tris.push_back(ParseTriangle(ss));
+				}
+				else if (identifier == "numweights") {
+					unsigned int nweights;
+					ss >> nweights;
+					mesh.weights.reserve(nweights);
+				}
+				else if (identifier == "weight") {
+					mesh.weights.push_back(ParsesWeight(ss));
+				}
+				// Check if the line contained the closing brace. This is done after parsing
+				// as the line might have the ending brace on it.
+				if (line.find("}") != std::string::npos) {
+					this->meshes_internal.push_back(std::move(mesh));
+					break;
+				}
 			}
 		}
-
-		return true;
 	}
 
-	void MD5Mesh::CalculateVertexPositions() {
-		if (this->meshes.size() < this->meshes_internal.size()) {
-			this->meshes.reserve(this->meshes_internal.size());
-			for (std::size_t i = this->meshes.size(); i < this->meshes_internal.size(); ++i) {
-				CreateMesh();
-			}
+	return true;
+}
+
+void MD5Mesh::CalculateVertexPositions() {
+	if (this->meshes.size() < this->meshes_internal.size()) {
+		this->meshes.reserve(this->meshes_internal.size());
+		for (std::size_t i = this->meshes.size(); i < this->meshes_internal.size(); ++i) {
+			CreateMesh();
 		}
+	}
 
-		for (std::size_t i = 0; i < this->meshes_internal.size(); ++i) {
-			Mesh* mesh = this->meshes[i];
-			InternalMesh& int_mesh = this->meshes_internal[i];
-			if (mesh->verts.size() < int_mesh.verts.size()) {
-				mesh->verts.resize(int_mesh.verts.size());
+	for (std::size_t i = 0; i < this->meshes_internal.size(); ++i) {
+		Mesh* mesh = this->meshes[i];
+		InternalMesh& int_mesh = this->meshes_internal[i];
+		if (mesh->verts.size() < int_mesh.verts.size()) {
+			mesh->verts.resize(int_mesh.verts.size());
+		}
+		for (std::size_t j = 0; j < int_mesh.verts.size(); ++j) {
+			VertexData vdata;
+
+			// Compute vertex position based on joint position.
+			for (std::uint8_t k = 0; k < int_mesh.verts[j].weight_count; ++k) {
+				Weight& weight =
```

Comment:
Fix me

Reply:
`int_mesh.verts[j]` shows up a lot, so does `this->joints[weight.joint]`, both could probably be converted to local variables.